teps/0104-tasklevel-resource-requirements.md

```diff
@@ -103,18 +103,15 @@ depending on its inputs. In addition, LimitRanges don’t distinguish between Te
 
 ### API Changes
 
-Augment the Task and TaskRun APIs with a resources field that allows the user to configure the resource requirements
-of a Task. An example Task is as follows.
+Augment the TaskRun API with a resources field that allows the user to configure the resource requirements
+of a Task. An example TaskRun is as follows.
 
 ```
 apiVersion: tekton.dev/v1beta1
-kind: Task
+kind: TaskRun
 metadata:
-  name: image-build-task
+  name: image-build-taskrun
 spec:
-  steps:
-  - name: step-1
-  - name: step-2
   resources:
     requests:
       memory: 1Gi
```

Comment:
thinking here we perhaps still will meet a "field overloading" problem for `TaskRun`:

```
type TaskRunSpec struct {
	// +optional
	Debug *TaskRunDebug `json:"debug,omitempty"`
	// +optional
	// +listType=atomic
	Params []Param `json:"params,omitempty"`
	// +optional
	Resources *TaskRunResources `json:"resources,omitempty"`
```
https://github.com/austinzhao-go/pipeline/blob/tep-0104-per-task-resources-requirements/pkg/apis/pipeline/v1beta1/taskrun_types.go#L44

Reply:
You're completely right -- addressed in https://github.com/tektoncd/community/pull/726